### ISSUE TITLE
Failed to start pusher: No client id specified

### ISSUE THREAD

**Issue by pogzie** [jprjr/multistreamer#26]
Im getting this message after I got this working a couple of days ago. Any idea on what this is?

**Comment by k1ck3r** [comment]
have the same issue only with twitch, and i suppose your issue is same.

**Comment by pogzie** [comment]
I tried un-associating the accounts and re-associating but it didnt work. For the sake of "fixing" it faster, I created a virtual machine image of a fresh install.

**Comment by k1ck3r** [comment]
actually if you re-authenticate twitch, everything is back to normal. already gave full explanattion in my opened issues regarding twitch authentication module. but seems Jp don't have enough time to look into it

**Comment by jprjr** [comment]
Hey sorry, you're right I haven't had too much time but I'll try to look into it now that I've got a break from work for the winter holidays.

**Comment by jprjr** [comment]
Identified the likely culprit - at some point, Twitch added expiration times to newly-generated tokens, which is why my installation never noticed this problem.

I'll have to write up token-refreshing code, similar to what the YouTube module does.

**Comment by pogzie** [comment]
Thanks good sir! Let us know if you have a new build so we can update!

**Comment by jprjr** [comment]
I've pushed 2 new branches - `fix-twitch` and `fix-twitch-legacy`, that *should* have the token-refreshing fix.

I say "should" because I just wrote it, and haven't had a token expire just yet for me to test the refresh, which is why it isn't in master. If anybody wants to try those branches out, feel free! I'll hopefully be confirming everything works and merging into master later tonight or tomorrow morning.
